client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-bulk-select/index.tsx

```diff
@@ -0,0 +1,90 @@
+import { useTranslate } from 'i18n-calypso';
+import { useContext } from 'react';
+import BulkSelect from 'calypso/components/bulk-select';
+import DashboardBulkActions from '../../dashboard-bulk-actions';
+import SitesOverviewContext from '../context';
+import type { SiteData } from '../types';
+
+import './style.scss';
+
+interface Props {
+	sites: Array< SiteData >;
+	isLoading: boolean;
+}
+
+export default function SiteBulkSelect( { sites, isLoading }: Props ) {
+	const translate = useTranslate();
+
+	const { selectedSites, setSelectedSites } = useContext( SitesOverviewContext );
+
+	const selectedSiteIds = selectedSites.map( ( site ) => site.blog_id );
+
+	const isAllChecked = ( sites: Array< SiteData > ) => {
+		return (
+			selectedSites.length > 0 &&
+			sites
+				.map( ( site ) => site.site.value.blog_id )
+				.every( ( item ) => selectedSiteIds.includes( item ) )
```

Comment:
Can we optimize this a bit by removing the `.map` operation?

```suggestion
			sites
				.every( ( site ) => selectedSiteIds.includes( site.site.value.blog_id  ) )
```